tests/udf_framework/test_create_event_handler_udf_mock.py

```diff
@@ -0,0 +1,143 @@
+from exasol_udf_mock_python.mock_meta_data import MockMetaData
+from exasol_udf_mock_python.udf_mock_executor import UDFMockExecutor
+from tests.udf_framework import mock_event_handlers
+
+
+def _udf_wrapper():
+    from exasol_udf_mock_python.udf_context import UDFContext
+    from exasol_advanced_analytics_framework.udf_framework.\
+        create_event_handler_udf import CreateEventHandlerUDF
+
+    udf = CreateEventHandlerUDF(exa)
+    def run(ctx: UDFContext):
+        udf.run(ctx)
+
+
+def _create_mock_data():
+    meta = MockMetaData(
+        script_code_wrapper_function=_udf_wrapper,
+        input_type="SET",
+        input_columns=[
+            Column("1", int, "INTEGER"),  # iter_num
+            Column("2", str, "VARCHAR(2000000)"),  # bucketfs_connection_name
+            Column("3", str, "VARCHAR(2000000)"),  # event_handler_class_name
+            Column("4", str, "VARCHAR(2000000)"),  # event_handler_module
+            Column("5", str, "VARCHAR(2000000)"),  # event_handler_parameters
+         ],
+        output_type="EMITS",
+        output_columns=[
+            Column("outputs", str, "VARCHAR(2000000)")
+        ],
+        is_variadic_input=True
+    )
+    return meta
+
+
+def test_event_handler_udf_with_one_iteration():
+    executor = UDFMockExecutor()
+    meta = _create_mock_data()
+
+    with TemporaryDirectory() as path:
+        bucketfs_connection = Connection(address=f"file://{path}/event_handler")
+        exa = MockExaEnvironment(
+            metadata=meta,
+            connections={"bucketfs_connection": bucketfs_connection})
+
+        input_data = (
+            0,
+            "bucketfs_connection",
+            "MockEventHandlerWithOneIteration",
+            "tests.udf_framework.mock_event_handlers",
+            "{}"
+        )
+        result = executor.run([Group([input_data])], exa)
+        for i, group in enumerate(result):
+            result_row = group.rows
+            assert len(result_row) == 4
+            is_finished = result_row[2][0]
+            final_result = result_row[3][0]
+            assert is_finished == "True"
+            assert final_result == str(mock_event_handlers.FINAL_RESULT)
+
+
+def test_event_handler_udf_with_two_iteration():
+    executor = UDFMockExecutor()
+    meta = _create_mock_data()
+
+    with TemporaryDirectory() as path:
+        bucketfs_connection = Connection(address=f"file://{path}/event_handler")
+        exa = MockExaEnvironment(
+            metadata=meta,
+            connections={"bucketfs_connection": bucketfs_connection})
+
+        input_data = (
+            0,
+            "bucketfs_connection",
+            "MockEventHandlerWithTwoIterations",
+            "tests.udf_framework.mock_event_handlers",
+            "{}"
+        )
+        result = executor.run([Group([input_data])], exa)
+        for i, group in enumerate(result):
+            result_row = group.rows
+            assert len(result_row) == 4 + len(mock_event_handlers.QUERY_LIST)
```

Comment:
use for the multiple assert pytest-assume https://github.com/astraw38/pytest-assume